### PR TITLE
Remove deprecated fields from attestation data

### PR DESCRIPTION
**Pull request by prestonvanloon**
This is mostly checkpoint changes. 

Also restores some (but not all) validation that was removed in #2867

### CODE REVIEW

File: beacon-chain/core/blocks/block_operations.go [hunk 1]
```diff
@@ -891,7 +924,25 @@ func ProcessTransfers(
...
 	transfers := body.Transfers
-
+	if uint64(len(transfers)) > params.BeaconConfig().MaxTransfers {
+		return nil, fmt.Errorf(
+			"number of transfers (%d) exceeds allowed threshold of %d",
+			len(transfers),
+			params.BeaconConfig().MaxTransfers,
+		)
+	}
+	// Verify there are no duplicate transfers.
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: we don't need this anymore, it's checked in `process_operation`:
https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/state/transition.go#L267

File: beacon-chain/core/blocks/block_operations.go [hunk 2]
```diff
@@ -659,6 +673,18 @@ func ProcessDeposits(
 ) (*pb.BeaconState, error) {
 	var err error
 	deposits := body.Deposits
+	// Verify that outstanding deposits are processed up to the maximum number of deposits.
+	maxDeposits := beaconState.Eth1Data.DepositCount - beaconState.Eth1DepositIndex
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: We don't need it here, it's checked as part of `process_operation`:
https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/state/transition.go#L262

File: beacon-chain/core/blocks/block_operations.go [hunk 3]
```diff
@@ -808,6 +834,13 @@ func ProcessVolundaryExits(
 ) (*pb.BeaconState, error) {
 	var err error
 	exits := body.VoluntaryExits
+	if uint64(len(exits)) > params.BeaconConfig().MaxVoluntaryExits {
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: we should short cut all these operation checks at the beginning of `process_operation` `https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/state/transition.go#L257`

so it doesn't get to the last process operation step and then realize it's a invalid block, it's wasteful

File: beacon-chain/core/blocks/block_operations.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/core/blocks/block_operations.go]: Why was it moved?

Comment (terencechain) [reply beacon-chain/core/blocks/block_operations.go]: It's not removed, it's moved to process_operation

File: beacon-chain/core/blocks/block_operations.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/core/blocks/block_operations.go]: Can we address these later? I was just trying to fix these types and I had test failures due to newly missing validation.

Comment (terencechain) [reply beacon-chain/core/blocks/block_operations.go]: sure, I was planning to make a PR that takes all the takes to `process_operation` level

File: beacon-chain/core/blocks/block_operations.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/core/blocks/block_operations.go]: so we don't have to process proposer slashings, attester slashings and attestations (1/2 of the process_block cycle) before realizing the deposit amount is incorrect

I can address this in my block operations PR if you need to merge this urgently

File: beacon-chain/core/blocks/block_operations.go (hunk 1 above)
Comment (0xKiwi) [reply beacon-chain/core/blocks/block_operations.go]: I've made a PR to handle this with tests as well #2900 👍

**Pull request merged**